### PR TITLE
Add support for native sDAI deposit

### PR DESCRIPTION
**Pull request by yivlad**
- Adds `nativeSDaiDeposit` action flavour 
- Introduces feature flag `VITE_DEV_NATIVE_ROUTES`
- Uses new action flavour when `VITE_DEV_NATIVE_ROUTES` is set to `1` and user is depositing DAI into sDAI on mainnet

### CODE REVIEW

File: packages/app/src/features/actions/flavours/native-sdai-deposit/useCreateNativeSDaiDepositHandler.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { useVaultDeposit } from '@/domain/tokenized-vault-operations/useVaultDeposit'
+
+import { PermitStore } from '../../logic/permits'
+import { ActionHandler } from '../../logic/types'
+import { mapWriteResultToActionState } from '../../logic/utils'
+import { NativeSDaiDepositAction } from './types'
+
+export interface UseCreateDepositHandlerOptions {
```
Comment (oskarvu) [inline packages/app/src/features/actions/flavours/native-sdai-deposit/useCreateNativeSDaiDepositHandler.ts:8]: `UseCreateNativeSDaiDepositHandlerOptions`

File: packages/app/src/features/actions/flavours/native-sdai-deposit/useCreateNativeSDaiDepositHandler.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  enabled: boolean
+  permitStore?: PermitStore
```
Comment (oskarvu) [inline packages/app/src/features/actions/flavours/native-sdai-deposit/useCreateNativeSDaiDepositHandler.ts:10]: Do you need it here?

**Review by oskarvu — COMMENTED** [review]
I would probably simplfiy name of `nativeSDaiDeposit` to just `sDaiDeposit`. Main reasons: - `native` reminds me of a native token,
- `deposit` already tells us it's not exchange, and that we will use sc.

File: packages/app/src/features/dialogs/savings/deposit/logic/useSavingsDepositDialog.ts [hunk 3]
```diff
@@ -69,7 +70,17 @@ export function useSavingsDepositDialog({
     form,
     marketInfo,
   })
-  const { swapInfo, swapParams } = useDepositIntoSavings({ formValues, marketInfo })
+
+  const useNativeRoutes =
+    import.meta.env.VITE_DEV_NATIVE_ROUTES === '1' &&
+    marketInfo.chainId === mainnet.id &&
```
Comment (oskarvu) [inline packages/app/src/features/dialogs/savings/deposit/logic/useSavingsDepositDialog.ts:76]: This won't use native routes on sandox mode, right?

Comment (yivlad) [reply packages/app/src/features/dialogs/savings/deposit/logic/useSavingsDepositDialog.ts:76]: Good catch

File: packages/app/src/features/actions/logic/useCreateActions.ts [hunk 4]
```diff
@@ -163,6 +164,22 @@ export function useCreateActions(objectives: Objective[]): Action[] {
 
         return [approveExchangeAction, exchangeAction]
       }
+
+      case 'nativeSDaiDeposit': {
+        const depositAction: NativeSDaiDepositAction = {
```
Comment (krzkaczor) [inline packages/app/src/features/actions/logic/useCreateActions.ts:169]: nit: flip order of declaration of these.

**Review by krzkaczor — COMMENTED** [review]
Counterintuitively, I wouldn't enable native routes in playwright tests now (and tweak e2e tests). I prefer e2e tests to test production environment. Can you revert these changes?

**Comment by krzkaczor** [comment]
@yivlad @oskarvu i think "native" sdai deposit is fine. Also future use with USDC is cleaner.

**Review by oskarvu — APPROVED** [review]

File: packages/app/src/features/actions/flavours/native-sdai-deposit/NativeSDaiActionRow.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      <ActionRow.Index index={index} />
+
+      <ActionRow.Icon path={assets.actions.exchange} actionStatus={status} />
+
+      <ActionRow.Title icon={<IconStack paths={tokenIconPaths} stackingOrder="last-on-top" />} actionStatus={status}>
+        Wrap {fromToken.symbol} into {toToken.symbol}
+      </ActionRow.Title>
+
+      <ActionRow.Description successMessage={successMessage} actionStatus={status} variant={variant}>
+        <UpDownMarker token={fromToken} value={action.value} direction={'up'} />
```
Comment (oskarvu) [inline packages/app/src/features/actions/flavours/native-sdai-deposit/NativeSDaiActionRow.tsx:37]: This should be `direction="down"` instead, right?

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**